Purchases/Purchasing/StoreKitAbstractions/StoreProduct.swift

```diff
@@ -184,6 +190,19 @@ public extension StoreProduct {
 
 }
 
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+public extension StoreProduct {
+    /// Finds the subset of ``discounts`` that's eligible for the current user.
+    /// - Note: if checking for eligibility for a `StoreProductDiscount` fails (for example, if network is down),
+    ///   that discount will fail silently and be considered not eligible.
+    /// - Warning: this method implicitly relies on ``Purchases`` already being initialized.
```

Comment:
sorry if i missed a discussion on this...but should we error if purchases isn't initialized?

Reply:
I could go either way. I think it's probably better to crash, just like calling `Purchases.shared` would?